drivers/ovsdriver.go

```diff
@@ -57,14 +61,38 @@ type OvsDriverConfig struct {
 	}
 }
 
+type OvsDriverOperState struct {
+	core.CommonState
+	// used to allocate port names. XXX: should it be user controlled?
+	CurrPortNum int `json:"currPortNum"`
```

Comment:
CurrPortNum is used for deriving the names of internal ports in ovs, so it needs to be restored to it's previous value in order to prevent drivers from reusing any existing ports. This is needed for driver to function correctly after a restart.

There is no explicit restore of Endpoint needed as part of `Init()` as the driver get's a replay of the configuration from the Plugin (as part of restore). Now on the replay, the checks in CreateEndpoint() ensure that the driver acts based on operational state, if any, it had created before which will ensure Driver continues to keep same port number etc for the endpoint.